### PR TITLE
feat(backToCart): let backToCart use the plugin config

### PR DESCRIPTION
**Pull request by spencerwahl**
## Link to issue number(s):
fgpv-vpgf/fgpv-vpgf#3061

## Summary of the issue:
Additional work surrounding moving backToCart to v3.

## Description of how this pull request fixes the issue:
Back to cart now recognizes a config snippet structured as:

```
"backToCart": {
    "catalogueUrl": <url>
}
```

## Testing:

The backToCart sample is now using the plguin config

https://spencerwahl.github.io/plugins/5aa1fb6ce228218bb36212678efce5199be63410/backToCart/samples/backtocart-index.html

-   [ ] Test specs are up-to-date & cover all changes/additions made by this PR.
-   [ ] `npm run test` passes locally.

## Documentation

-   [x] Documentation is up-to-date - any changes or additions have been noted
-   [ ] `changelog.md` has been updated

## Checklist

-   [x] PR has only one commit (squash otherwise)
-   [x] Commit message is descriptive

---
This change is [Reviewable](https://reviewable.io/reviews/fgpv-vpgf/plugins/47)

### CODE REVIEW

File: backToCart/index.ts [hunk 1]
```diff
@@ -26,7 +25,7 @@ export default class BackToCart {
     getCatalogueUrl(): string {
         if (!this.template) {
             console.warn('<Back to Cart> Trying to get URL before template is set');
-            return '';
+            return;
```
Comment (milesap) [inline backToCart/index.ts]: Method declares a return value of `string` but it seems this returns `undefined`?

File: backToCart/index.ts [hunk 2]
```diff
@@ -36,12 +35,24 @@ export default class BackToCart {
...
             // save bookmark in local storage so it is restored when user returns
             sessionStorage.setItem(this.api.id, this._RV.getBookmark());
             window.location.href = this.getCatalogueUrl();
         };
     }
 
+    /**
+     * Auto called by RAMP startup, checks for a catalogueUrl in the config and sets the template if so
+     */
+    preInit(pluginConfig: any): void {
```
Comment (milesap) [inline backToCart/index.ts]: Define the `pluginConfig` interface instead of using `any` type.

Comment (spencerwahl) [reply backToCart/index.ts]: Done

File: backToCart/index.ts (hunk 1 above)
Comment (spencerwahl) [reply backToCart/index.ts]: switched to have both return values, value of undefined used to cancel button click callback

**Pull request merged**